### ISSUE TITLE
Assumed coordinate systems for draw_rectangle and draw_convex_polygons are different

### ISSUE THREAD

**Issue by Parskatt** [kornia/kornia#1641]
### Describe the bug

Draw rectangle assumes (by casting coordinates to int), a coordinate system with the center of the top-left pixel at [0.5,0.5], while draw_convex_polygons assumes [0,0] at the center of the top-left. While the assumed coordinate system is noted in draw_convex_polygons, it is not in draw_rectangle. This means that the edges of the draw objects will not be consistent for the same input, and may be confusing.

### Reproduction steps

```bash
For example adding the test:


    def test_rectangle_tiny_shift(self, device, dtype):
        b, c, h, w = 1, 3, 500, 500
        im = torch.zeros(b, c, h, w, device=device, dtype=dtype)
        color = torch.tensor([1, 1, 1], device=device, dtype=dtype)[None].expand(b, c)
        pts = torch.tensor([[[50, 50], [200, 50], [200, 250], [50, 250]]], device=device, dtype=dtype)+0.0001
        poly_im = draw_convex_polygons(im.clone(), pts, color)
        rect = torch.cat((pts[..., 0, :], pts[..., 2, :]), dim=-1)[:, None]
        rect_im = draw_rectangle(im.clone(), rect, color[:, None], fill=True)
        assert torch.allclose(rect_im, poly_im)

will fail.
```

### Expected behavior

The drawing functions ought to have the same coordinate system.

### Environment

```shell
-
```

**Comment by Parskatt** [comment]
I think this problem is also prevalent in multiple other parts of kornia, where the coordinate systems used are not explicit.
create_meshgrid e.g. assumes a [0,0] center of top-left pixel, but this is not specified.

Assuming [0.5,0.5] for [0,0] and vice-versa may introduce subtle errors especially when pixel precision is required, so I think it's important that we are very clear with what coordinate system is assumed.

**Comment by edgarriba** [comment]
@Parskatt I agree -- we had this discussion with @ducha-aiki while deciding similar conventions for warping. In pytorch we can control that with the `align_corners` (from `grid_sample`) to keep backward compatibility and align with opencv we are setting align corners to false by default which mean I believe [0,0] center.

**Comment by Parskatt** [comment]
@edgarriba I'm pretty sure that `align_corners = False` implies [0.5,0.5] center of top-left.
I've had so many issues with this when doing geometric matching haha.

**Comment by edgarriba** [comment]
Oh, okay, then I’d keep that standard. Unless we have another preference. In drawing api my guess is that we are using heavily indexing so using subpixel precision won’t easy at all with the current approach.

**Comment by Parskatt** [comment]
Yeah, unless we do anti-aliased lines there is only 0.5 pixel resolution since the coloring is binary. I think I will make a pull request later to fix draw polygon to assume [0.5,0.5] indexing so that it is consistent with draw_rectangle.

**Comment by edgarriba** [comment]
cool, yeah sounds good. Would be also great to small a small colab showing the different drawing functions. (basically plotting the tests)

**Comment by Parskatt** [comment]
@edgarriba you mean in the examples section? I might be able to later next week.

**Comment by edgarriba** [comment]
we have this: https://github.com/kornia/tutorials

**Comment by Parskatt** [comment]
Great! I'll make a larger write-up later, and maybe include non-convex polygons and quadratic bezier curves there as well.

**Comment by edgarriba** [comment]
If anyone is interested in this, feel free to re-open the ticket. Thanks!
